pkg/initializer/initializer.go

```diff
@@ -56,9 +56,15 @@ func (e *EtcdInitializer) Initialize(mode validator.Mode, failBelowRevision int6
 		m := member.NewMemberControl(e.Config.EtcdConnectionConfig)
 		isEtcdMemberPresent, err = m.IsMemberInCluster(ctx)
 		if !isEtcdMemberPresent && err == nil {
-			retry.OnError(retry.DefaultBackoff, func(err error) bool {
-				return err != nil
-			}, func() error {
+			retry.OnError(retry.DefaultBackoff, errors.IsErrNotNil, func() error {
+				// Additional safety check before adding a learner
+				if _, err := os.Stat(e.Config.RestoreOptions.Config.RestoreDataDir); err == nil {
+					if err := os.RemoveAll(filepath.Join(e.Config.RestoreOptions.Config.RestoreDataDir)); err != nil {
+						return fmt.Errorf("failed to remove directory %s with err: %v", e.Config.RestoreOptions.Config.RestoreDataDir, err)
```

Comment:
If you try and remove the data directory and it returns an error then you will end up trying infinitely, right? Would unlimited number of retries in this case help?

Reply:
I think, it would retry for DefaultBackoff period

Reply:
yes, it will retry for DefaultBackoff period